### PR TITLE
test(e2e,playwright): Map

### PR DESCRIPTION
**Pull request by Riya267**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

 #51705

Snapshot:-
9323-riya267-freecodecamp-5jzi1y15q58 ws-us105 gitpod io_

### CODE REVIEW

**Comment by Riya267** [comment]
camperbot cant code

File: e2e/map.spec.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    await expect(
+      page.getByText(translations.landing['interview-prep-heading'])
+    ).toBeVisible();
+  });
+
+  test('verifies all the certifications links rendered correctly', async ({
+    page,
+    browserName
+  }) => {
+    test.skip(browserName === 'webkit');
```
Comment (naomi-lgbt) [inline e2e/map.spec.ts:69]: We shouldn't need to skip webkit here. The map appears whether authenticated or not.

File: e2e/map.spec.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+test.beforeEach(async ({ page }) => {
+  await page.goto('/learn');
+});
+
+test.afterEach(async ({ page }) => {
+  await page.close();
+});
+
+const superBlocksWithLinks = [
```
Comment (naomi-lgbt) [inline e2e/map.spec.ts:12]: I wonder if we should pull this from the shared config.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: e2e/map.spec.ts (hunk 1 above)
Comment (Riya267) [reply e2e/map.spec.ts]: yes right, I have removed it .

File: e2e/map.spec.ts (hunk 2 above)
Comment (Riya267) [reply e2e/map.spec.ts:17]: done

File: e2e/map.spec.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  superBlockOrder
+} from '../shared/config/superblocks';
+test.beforeEach(async ({ page }) => {
+  await page.goto('/learn');
+});
+
+test.afterEach(async ({ page }) => {
+  await page.close();
+});
+
```
Comment (naomi-lgbt) [inline e2e/map.spec.ts:16]:
```suggestion
```

We actually don't need this. Playwright closes the page automatically.

File: e2e/map.spec.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      const superblockLink = page.getByRole('link', {
+        name: intro[superBlocksWithLinks[i]].title
+      });
+      expect(await superblockLink.getAttribute('href')).toBe(
+        `/learn/${superBlocksWithLinks[i]}/`
+      );
+      await superblockLink.click();
+    }
+  });
+});
```
Comment (naomi-lgbt) [inline e2e/map.spec.ts:53]:
```suggestion
test.describe('Map Component', () => {
  test('should render correctly', async ({ page }) => {
    await expect(
      page.getByText(translations.landing['core-certs-heading'])
    ).toBeVisible();
    await expect(
      page.getByText(translations.landing['professional-certs-heading'])
    ).toBeVisible();
    await expect(
      page.getByText(translations.landing['interview-prep-heading'])
    ).toBeVisible();
    const curriculumBtns = page.getByTestId('curriculum-map-button');
    await expect(curriculumBtns).toHaveCount(15);
    for (let i = 0; i < superBlocksWithLinks.length; i++) {
      const superblockLink = page.getByRole('link', {
        name: intro[superBlocksWithLinks[i]].title
      });
      expect(await superblockLink.getAttribute('href')).toBe(
        `/learn/${superBlocksWithLinks[i]}/`
      );
      await superblockLink.click();
    }
  });
});
```

We could also probably merge these into one test.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: e2e/map.spec.ts (hunk 3 above)
Comment (Riya267) [reply e2e/map.spec.ts]: done

File: e2e/map.spec.ts (hunk 4 above)
Comment (Riya267) [reply e2e/map.spec.ts]: done

**Review by naomi-lgbt — APPROVED** [review]

**Review by gikf — APPROVED** [review]

**Review by moT01 — APPROVED** [review]
LGTM @Riya267 🎉 Thanks for contributing to freeCodeCamp and happy hacktoberfest 🎉

**Pull request merged**